packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx

```diff
@@ -154,11 +162,11 @@ class ToggleMenuDrawer extends React.Component {
 
     render() {
         const all_routes_config = getAllRoutesConfig();
-        const subroutes_config = [].concat(...all_routes_config.map(i => i.routes || []));
-        const allowed_routes = [routes.reports, routes.account, routes.cashier];
-        const routes_config = allowed_routes
-            .map(path => all_routes_config.find(r => r.path === path) || subroutes_config.find(r => r.path === path))
-            .filter(route => route);
+        const primary_routes = [routes.reports, routes.account, routes.cashier];
+        const primary_routes_config = this.getFilteredRoutesConfig(all_routes_config, primary_routes);
```

Comment:
do we need to call all these functions on every render or is there another way? 🤔

Reply:
Now that we get all the route configs from core, we can move this logic componentDidMount